core/clients/key_flow.go

```diff
@@ -294,26 +327,44 @@ func (c *KeyFlow) parseTokenResponse(res *http.Response) error {
 		return fmt.Errorf("received bad response from API")
 	}
 	if res.StatusCode != http.StatusOK {
-		return fmt.Errorf("received: %+v", res)
+		body, err := io.ReadAll(res.Body)
+		if err != nil {
+			// Fail silently, omit body from error
+			// We're trying to show error details, so it's unnecessary to fail because of this err
+			body = []byte{}
+		}
+		return &oapierror.GenericOpenAPIError{
```

Comment:
Changed to be an `oapierror.GenericOpenAPIError` so the token refresh can fetch the status code